Tools/CISupport/ews-build/steps.py

```diff
@@ -362,6 +362,28 @@ def update_pr(self, pr_number, title, description, base=None, head=None, reposit
             return False
         return True
 
+    def close_pr(self, pr_number, repository_url=None):
+        api_url = GitHub.api_url(repository_url)
+        if not api_url:
+            return False
+
+        update_url = f'{api_url}/pulls/{pr_number}'
+        try:
+            username, access_token = GitHub.credentials()
+            auth = HTTPBasicAuth(username, access_token) if username and access_token else None
+            response = requests.request(
+                'POST', update_url, timeout=60, auth=auth,
+                headers=dict(Accept='application/vnd.github.v3+json'),
+                json=dict(state='closed'),
+            )
+            if response.status_code // 100 != 2:
+                self._addToLog('stdio', f"Failed to close PR {pr_number}. Unexpected response code from GitHub: {response.status_code}\n")
+                return False
+        except Exception as e:
+            self._addToLog('stdio', f"Error in closing PR {pr_number}\n")
```

Comment:
do we want to print `e` here?

Reply:
We should, will add

Reply:
Actually, we haven't logged a similar exception in any of the other functions in `GitHubMixin`, I will propose a change that does this, but we should do all of them all at the same time.